File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        visited = set()
        for i, num in enumerate(nums):
            complement = target - num
            if complement not in visited:
                visited.add(num)
                continue
            for j in range(i):
                if nums[j] == complement:
                    return [i, j]

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_set<int> uset;
        vector<int> ans;
        for (int i = 0; i < nums.size(); ++i) {
            if (uset.find(target - nums[i]) != uset.end()) {
                ans.push_back(i);
                for (int j = 0; j < nums.size(); ++j) {
                    if (nums[j] == target - nums[i]) {
                        ans.push_back(j);
                        break;
                    }
                }
                break;
            }
            uset.insert(nums[i]);
        }
        return ans;
    }
};